src/main/java/org/apache/accumulo/testing/healthprobe/Monitor.java

```diff
@@ -0,0 +1,203 @@
+        scanner.setRange(range);
+        if (opts.batch_size > 0) {
+          scanner.setBatchSize(opts.batch_size);
+        }
+        try {
+          long startTime = System.nanoTime();
+          int count = consume(scanner);
+          long stopTime = System.nanoTime();
+          MDC.put("StartTime", String.valueOf(startTime));
+          MDC.put("TabletIndex", String.valueOf(tablet_index_generator));
+          MDC.put("TableName", String.valueOf(opts.tableName));
+          MDC.put("TotalTime", String.valueOf((stopTime - startTime)));
+          MDC.put("StartRow", String.valueOf(range.getStartKey()));
+          MDC.put("EndRow", String.valueOf(range.getEndKey()));
+          MDC.put("TotalRecords", String.valueOf(count));
+
+          log.info("SCN starttime={} tabletindex={} tablename={} totaltime={} totalrecords={}", startTime,
+              tablet_index_generator, opts.tableName, (stopTime - startTime), count);
+          if (scannerSleepMs > 0) {
+            sleepUninterruptibly(scannerSleepMs, TimeUnit.MILLISECONDS);
+          }
+        } catch (Exception e) {
+          log.error(String.format(
+              "Exception while scanning range %s. Check the state of Accumulo for errors.", range), e);
+        }
+      }
+    }
+  }
+
+  public static int consume(Iterable<Entry<Key,Value>> iterable) {
+    Iterator<Entry<Key,Value>> itr = iterable.iterator();
+    int count = 0;
+    while (itr.hasNext()) {
+      Entry<Key,Value> e = itr.next();
+      Key key = e.getKey();
+      String readRow= key.getRow() + " " + key.getColumnFamily() + " " + key.getColumnQualifier() + " " + e.getValue();
+      count++;
+    }
+    return count;
+  }
+
+  public static long genLong(long min, long max, Random r) {
+    return ((r.nextLong() & 0x7fffffffffffffffL) % (max - min)) + min;
+  }
+
+  static byte[] genRow(long min, long max, Random r) {
+    return genRow(genLong(min, max, r));
+  }
+
+  public static byte[] genRow(long rowLong) {
+    return FastFormat.toZeroPaddedString(rowLong, 16, 16, EMPTY_BYTES);
+  }
+
+  public static Text genMaxRow(Text splitEnd) {
+    String padded = splitEnd.toString() + "0000000000000000".substring(splitEnd.toString().length());
+    Text maxRow = new Text(genRow( Long.parseLong(padded.trim(), 16) - 1l));
+    return maxRow;
+  }
+
+  public static Text genMinRow(Text splitStart) {
+    String padded = splitStart.toString()
+        + "0000000000000000".substring(splitStart.toString().length());
+    Text minRow = new Text(genRow(Long.parseLong(padded.trim(), 16)));
+    return minRow;
+  }
+
+  public static Range pickRange(TableOperations tops, String table, Random r)
+      throws TableNotFoundException, AccumuloSecurityException, AccumuloException {
+
+    ArrayList<Text> splits = Lists.newArrayList(tops.listSplits(table));
+    if (splits.isEmpty()) {
+      return new Range();
+    } else {
+      int index = r.nextInt(splits.size());
```

Comment:
What value does this provide over selecting a random split and scanning the head of that split? 

This seems to be testing the ability to descend the indices of compressed blocks then choosing random keys within there. Could we not remove the complexity here by simply scanning the beginning of the range? That could also make this a more general solution for different data models. @keith-turner would you disagree with this?

Reply:
@phrocker @keith-turner I agree. It would be nice to make this more general purpose & schema agnostic , but at the same time I would like to retain the functionality that alllows us to scan configurable number of rows. Can you throw some light on what would be the best way (function) to scan beginning of the range without comprising on ability to read configurable number of rows? A quick code blurb that elaborates more as a starter would be helpful.

Reply:
@phrocker @keith-turner Please take a look at the latest commit. Schema Agnostic , Scans configurable number of rows from beginning of the tablet. Caches tablet lookup information and reuses on later iterations.